lib/dns.js

```diff
@@ -213,6 +215,7 @@ function resolver(bindingName) {
       throw new ERR_INVALID_CALLBACK();
     }
 
+    name = toASCII(name, true);
```

Comment:
this function can theoretically throw in https://github.com/nodejs/node/blob/2c0a75118cd6a2eaf6b45fe8c67336f44c86ab0f/lib/punycode.js#L335. Do we need to handle it?

Reply:
idk, if so, shouldn't we also handle it here to be lenient whether it uses icu or punycode?

Reply:
`url.parse` is synchronous so it's less of an issue. Here the error could only lead to an uncaught exception.

Reply:
Updated to ignore exceptions. Thanks